sources/packages/load-test/src/utils/auth/token-helpers.ts

```diff
@@ -0,0 +1,101 @@
+import http from "k6/http";
+import { AuthorizedParties } from "../sims-api/sims-api";
+import { KEY_CLOAK_TOKEN_URL } from "../../../config.env";
+import { decodeJWT } from "./jwt-helpers";
+import { TokenCacheResponse, UserPasswordCredential } from "./models";
+
+/**
+ * Default amount of time to a token be expired and be
+ * considered candidate to be renewed.
+ */
+const TOKEN_RENEWAL_SECONDS = 30;
+
+/**
+ * Caches the token already acquired.
+ */
+const tokenCache: Record<string, TokenCacheResponse> = {};
+
+/**
+ * Define if a token need to be renewed based in the jwt exp property.
+ * @param jwtExp exp token property that defines the expiration.
+ * @param maxSecondsToExpired Amount of seconds before the token expires that should be considered to renew it.
+ * For instance, if a token has 10min expiration and the parameter is defined as 30, the token will
+ * be renewed the result will be true if the token was acquired 9min30s ago or if it is already expired.
+ * @returns true if the token is expired or about to expire based on the property maxSecondsToExpired.
+ */
+function needRenewJwtToken(jwtExp: number, maxSecondsToExpired = 0): boolean {
+  const jwtExpMilliseconds = jwtExp * 1000; // Convert to milliseconds.
+  const maxSecondsToExpiredMilliseconds = maxSecondsToExpired * 1000; // Convert to milliseconds.
+  const expiredDate = new Date(
+    jwtExpMilliseconds - maxSecondsToExpiredMilliseconds
+  );
+  return new Date() > expiredDate;
+}
+
+/**
+ * Get a token for a Keycloak client optionally using a different
+ * cache key when there are variations of a token under the same client.
+ * For instance, for the Ministry, a token for each group can be acquired.
+ * @param authorizedParty Keycloak client.
+ * @param credentials user and password to acquire the token (the user must have
+ * a hardcoded password to bypass the Keycloak identity provider).
+ * @param options.
+ * - `uniqueTokenCache` different cache key when there are variations of
+ * a token under the same client. If not provided the user name will be
+ * used instead.
+ * @returns a token that can be used to perform API calls. Intended to be
+ * used only for the load tests.
+ */
+export function getCachedToken(
```

Comment:
This can get a new token as well. Can we use a name suitable for both scenarios.